### PR TITLE
Add Pasteup2016 release process to frontend

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

@sammorrisdesign and I are in the process of retiring the current guss/pasteup repos and packages and centralising everything we use from them in the frontend codebase itself.

We are the main users of them, and the general source of truth for them. This will mean we won't have to jump through the release hoops to update the site (which some people have actually stopped doing anyway).

As part of this, we are adding [a new package on NPM](https://www.npmjs.com/package/pasteup) which will distribute Pasteup2016/next/2.0/🌟, and which will be published directly from frontend.

We will retire those repos/packages gracefully once this is all in place.
## Do I need to do anything about this?

No, not until you hear more. We'll send an email when it's in use, this is just preparing the ground.
## What is the value of this and can you measure success?

Hopefully, making edits and managing the components will become massively simpler, while 3rd parties are just as able to make use of them.
## Does this affect other platforms - Amp, Apps, etc?

Possibly, yes.
## Screenshots

screen shot 2016-04-08 at 17 18 21

screen recording 2016-04-08 at 04 47 pm

screen recording 2016-04-08 at 04 47 pm-2
## Request for comment

@sammorrisdesign 
## 

_Does this PR meet the [contributing guidelines](https://github.com/guardian/frontend/blob/issue_pr_templates/.github/CONTRIBUTING.md#submission)?_

### CODE REVIEW

File: static/src/stylesheets/pasteup/publish.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        'init.author.name=Guardian developers',
+        'init.author.email=dotcom.engineers@theguardian.com',
+        '//registry.npmjs.org/:_authToken=' + authToken,
+        'registry=https://registry.npmjs.org/'
+    ].join('\n'));
+}
+
+function getReleaseType() {
+    return new Promise(function(resolve, reject) {
+        require("inquirer").prompt([{
```
Comment (jamespamplin) [inline static/src/stylesheets/pasteup/publish.js]: Slight style suggestion, I'd usually put all requires / imports at the top of the file. Makes it easier to see what dependencies are used per module.

File: static/src/stylesheets/pasteup/publish.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+setCredentials(npmCredentials)
+    .then(getReleaseType)
+    .then(release)
+    .then(function (response) {
+        megalog.log('Released a new version of `pasteup`:\n\n`' + response + '`');
+        return del(['node_modules', '.npmrc']);
+    })
+    .catch(function (e) {
+        console.error(e.stack);
+        process.exit(0);
```
Comment (jamespamplin) [inline static/src/stylesheets/pasteup/publish.js]: Any reason to `exit(0)` here? Should have `exit(1)` for any errors if you can.

**Comment by jamespamplin** [comment]
:+1: looks good to me

File: grunt-configs/eslint.js [hunk 3]
```diff
@@ -10,6 +10,9 @@ module.exports = function () {
         'grunt-configs': [
             'grunt-configs/**/*.js'
         ],
+        'pasteup': [
+            'static/src/stylesheets/pasteup/**/*.js'
+        ],
```
Comment (OliverJAsh) [inline grunt-configs/eslint.js]: Won't this already be covered by the `static/src/**/*.js` glob below?

Comment (sndrs) [inline grunt-configs/eslint.js]: good point, updated in #12536

**Pull request merged**